openedx/features/personalized_learner_schedules/call_to_action.py

```diff
@@ -96,13 +100,43 @@ def _log_past_due_warning(name):
     @staticmethod
     def _make_reset_deadlines_cta(xblock):
         from lms.urls import RESET_COURSE_DEADLINES_NAME
-        return {
+        request = get_current_request()
+        is_learning_mfe = request and is_request_from_learning_mfe(request)
+        course_key = xblock.scope_ids.usage_id.context_key
+
+        cta_data = {
             'link': reverse(RESET_COURSE_DEADLINES_NAME),
             'link_name': _('Shift due dates'),
             'form_values': {
-                'course_id': xblock.scope_ids.usage_id.context_key,
+                'course_id': course_key,
             },
             'description': _('To participate in this assignment, the suggested schedule for your course needs '
                              'updating. Don’t worry, we’ll shift all the due dates for you and you won’t lose '
                              'any of your progress.'),
+
         }
+
+        if is_learning_mfe:
+            if course_home_mfe_dates_tab_is_active(course_key):
+                body_link = get_microfrontend_url(course_key=course_key, view_name='dates')
+            else:
+                body_link = '{}{}'.format(settings.LMS_ROOT_URL, reverse('dates', args=[six.text_type(course_key)]))
+
+            cta_data['event_data'] = {
+                'body': {
+                    'text': _('View all dates'),
+                    'link': body_link,
+                },
+                'event_name': 'post_event',
+                'header': {
+                    'text': _('Your due dates have been successfully shifted to help you stay on track.'),
+                },
```

Comment:
So what about an error message if the POST *doesn't* work? Like "Could not shift your dates".

Header and body here could maybe be marked as like, success_header and success_body? And add an error_ version of both?

Or we could return the error / success message as part of the JSON in the target endpoint. That way, the endpoint, which would have more info about what went wrong or right could return the right string. Though that approach means that all CTAs would have to have similar JSON return formats, which maybe isn't wise to commit to?